stetho/src/main/java/com/facebook/stetho/inspector/elements/android/AndroidDOMProvider.java

```diff
@@ -129,4 +157,140 @@ public View getHighlightingView(Object element) {
 
     return highlightingView;
   }
+
+  private List<Window> getWindows() {
+    ArrayList<Window> windows = new ArrayList<Window>();
+
+    Descriptor appDescriptor = getDescriptor(mApplication);
+    if (appDescriptor == null) {
+      return windows;
+    }
+
+    int activityCount = appDescriptor.getChildCount(mApplication);
+    for (int ai = 0; ai < activityCount; ++ai) {
+      final Activity activity = (Activity)appDescriptor.getChildAt(mApplication, ai);
+      Descriptor activityDescriptor = getDescriptor(activity);
+      if (activityDescriptor == null) {
+        continue;
+      }
+
+      int windowCount = activityDescriptor.getChildCount(activity);
+      for (int wi = 0; wi < windowCount; ++wi) {
+        final Window window = (Window)activityDescriptor.getChildAt(activity, wi);
+        windows.add(window);
+      }
+    }
+
+    return windows;
+  }
+
+  public final class InspectModeHandler {
+    private final Runnable mEnableRunnable;
+    private final Runnable mDisableRunnable;
+    private List<View> mOverlays;
+
+    private final Predicate<View> mViewSelector = new Predicate<View>() {
+      @Override
+      public boolean apply(View view) {
+        return !(view instanceof DOMHiddenView);
+      }
+    };
+
+    public InspectModeHandler() {
+      mEnableRunnable = new Runnable() {
```

Comment:
nit: I consider this lambda-style pattern to be most readable if you logically think of this expression as a function definition.  In that way, it would read:
```java
public void enable() { ... }
  
private void enableOnUiThread() { ... }
  
private final Runnable mEnableRunnable = () -> {
  // code here...
};
```

Yes it technically violates that member variables go up top, but the pattern really is about a defining a function and referring to it as a pointer.  In C/C++, for instance, do exactly that making it really unintuitive for you to declare the method inside a function body as a nested function.